### PR TITLE
fix __doc__ is None in optimized mode

### PR DESCRIPTION
**Pull request by Irvenae**
# Pull Request Check List

This is just a friendly reminder about the most common mistakes.  Please make sure that you tick all boxes.  But please read our [contribution guide](https://www.structlog.org/en/latest/contributing.html) at least once, it will save you unnecessary review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing left to do.

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.
    - [x] New functions/classes have to be added to `docs/api.rst` by hand.
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).  Find the appropriate next version in our [``__init__.py``](https://github.com/hynek/structlog/blob/main/src/structlog/__init__.py) file.
- [x] Documentation in `.rst` files is written using [semantic newlines](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [x] Changes (and possible deprecations) are documented in the [changelog](https://github.com/hynek/structlog/blob/main/CHANGELOG.rst).

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

Not sure how a specific test can be launched with this env variable (PYTHONOPTIMIZE=2)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -24,7 +24,7 @@ Deprecations:
 Changes:
 ^^^^^^^^
 
-*none*
+- Fixed ``structlog`` when running in optimized mode (PYTHONOPTIMIZE=2).
```
Comment (hynek) [inline CHANGELOG.rst:27]:
```suggestion
- Fixed import when running in optimized mode (``PYTHONOPTIMIZE=2``).
   `#373 <https://github.com/hynek/structlog/pull/373>`_
```

File: src/structlog/__init__.py [hunk 2]
```diff
@@ -45,7 +45,11 @@
 __version__ = "21.4.0.dev0"
 
 __title__ = "structlog"
-__description__ = __doc__.strip()
+if __doc__ is None:
+    __description__ = ""
```
Comment (hynek) [inline src/structlog/__init__.py:49]:
```suggestion
    # __doc__ is None when running with PYTHONOPTIMIZE=2 / -OO
    __description__ = ""  # pragma: no cover
```

**Review by hynek — CHANGES_REQUESTED** [review]
Yeah you can't test that. Let's just exclude it from coverage.

Thanks!

**Comment by Irvenae** [comment]
Thanks for the fast review 🚀 💯

**Comment by hynek** [comment]
lolsob looks like a) i've screwed up the indent in the changelog and b) mypy doesn't know about `-OO`. I'll just merge and fiddle until it works – thanks!

**Comment by hynek** [comment]
JFTR, I've actually managed to build a test case: 1d66f67fbb519282db670999250cdd99474470be 😅

**Comment by hynek** [comment]
FYI 21.4 is on PyPI

**Pull request merged**